### PR TITLE
Add setting for .lua extension

### PR DESCRIPTION
**Pull request by Wilma456**
edit will only add .lua extension, when edit.extension is true.

### CODE REVIEW

File: src/main/resources/assets/computercraft/lua/rom/programs/edit.lua [hunk 1]
```diff
@@ -14,8 +14,10 @@ if fs.exists( sPath ) and fs.isDir( sPath ) then
 end
 
 -- Create .lua files by default
-if not fs.exists( sPath ) and not string.find( sPath, "%." ) then
-	sPath = sPath .. ".lua"
+if settings.get( "edit.extension" ) then
+   if not fs.exists( sPath ) and not string.find( sPath, "%." ) then
```
Comment (SquidDev) [inline src/main/resources/assets/computercraft/lua/rom/programs/edit.lua]: Just combine these into one if statement.

Comment (SquidDev) [inline src/main/resources/assets/computercraft/lua/rom/programs/edit.lua]: You're also using 3 spaces here and a tab character then 3 spaces on the next line. Seeing as the rest of the file uses 4 spaces, you should probably change this. 

This isn't the first time I've said this on your PRs, so can you please turn whitespace highlighting on in your editor - having a consistent whitespace makes it easier to read and edit the code in the future.

File: src/main/resources/assets/computercraft/lua/rom/help/settings.txt [hunk 2]
```diff
@@ -11,6 +11,7 @@ Default Settings:
 shell.autocomplete - enables auto-completion in the Shell.
 lua.autocomplete - enables auto-completion in the Lua program.
 edit.autocomplete - enables auto-completion in the Edit program.
+edit.extension - edit will add .lua extension
```
Comment (SquidDev) [inline src/main/resources/assets/computercraft/lua/rom/help/settings.txt]: Probably should read `edit will add .lua when no extension is specified` or something.

**Comment by dan200** [comment]
May I propose a different design for this:

Call the setting "edit.default_extension"
Then, when attempting to edit a file which doesn't exist, and whose name does not already include an extension:
If the setting is nil, or the empty string, do nothing.
If the setting is any other string, append "." plus the string to the filename.
Obviously, "lua" would be the default value.

**Comment by Wilma456** [comment]
@dan200  I have done this now

File: src/main/resources/assets/computercraft/lua/bios.lua [hunk 3]
```diff
@@ -817,7 +817,8 @@ end
 settings.set( "shell.allow_startup", true )
 settings.set( "shell.allow_disk_startup", (commands == nil) )
 settings.set( "shell.autocomplete", true )
-settings.set( "edit.autocomplete", true )
+settings.set( "edit.default_extension", ".lua" )
```
Comment (dan200) [inline src/main/resources/assets/computercraft/lua/bios.lua]: This should be "lua" not ".lua"

File: src/main/resources/assets/computercraft/lua/bios.lua [hunk 4]
```diff
@@ -817,7 +817,8 @@ end
...
+settings.set( "edit.extension", true )
```
Comment (dan200) [inline src/main/resources/assets/computercraft/lua/bios.lua]: Remove this setting. Extensionless behaviour should be indicated by setting default_extension to ""

File: src/main/resources/assets/computercraft/lua/rom/programs/edit.lua [hunk 5]
```diff
@@ -15,7 +15,7 @@ end
 
 -- Create .lua files by default
 if not fs.exists( sPath ) and not string.find( sPath, "%." ) then
-	sPath = sPath .. ".lua"
+    sPath = sPath..settings.get("edit.default_extension")
```
Comment (dan200) [inline src/main/resources/assets/computercraft/lua/rom/programs/edit.lua]: Only do this line if the setting is non-empty, and add the dot automatically

Comment (Restioson) [inline src/main/resources/assets/computercraft/lua/rom/programs/edit.lua]: Also, whitespace between `h`, `..`, and `s`

**Comment by dan200** [comment]
Will merge once above changes are made @Wilma456

File: src/main/resources/assets/computercraft/lua/bios.lua [hunk 6]
```diff
@@ -817,7 +817,8 @@ end
 settings.set( "shell.allow_startup", true )
 settings.set( "shell.allow_disk_startup", (commands == nil) )
 settings.set( "shell.autocomplete", true )
-settings.set( "edit.autocomplete", true )
```
Comment (Wojbie) [inline src/main/resources/assets/computercraft/lua/bios.lua]: Wait a second why are you removing edit.autocomplete in this line? This is still a valid setting and should not be removed.

Comment (dan200) [inline src/main/resources/assets/computercraft/lua/bios.lua]: Can't believe I didn't notice this. What a mess

**Comment by Wilma456** [comment]
@dan200  Now done!

**Comment by dan200** [comment]
Once again we had to change every single line in what should of been a trivial commit... butt nevertheless this is ready to merge

**Pull request merged**